examples/flutter_login/lib/authentication/bloc/authentication_bloc.dart

```diff
@@ -19,9 +20,17 @@ class AuthenticationBloc
         _authenticationRepository = authenticationRepository,
         _userRepository = userRepository,
         super(const AuthenticationState.unknown()) {
-    _authenticationStatusSubscription = _authenticationRepository.status.listen(
-      (status) => add(AuthenticationStatusChanged(status)),
-    );
+    _authenticationStatusSubscription =
+        _authenticationRepository.status.listen((status) {
+      final profile = HydratedBloc.storage?.read('ProfileBloc') as Map;
```

Comment:
We should not need to read the hydrated storage directly and should instead allow hydrate bloc instances to manage their cache automatically. I suggest we convert the `AuthenticationBloc` to a hydrated bloc rather than introducing a new `ProfileBloc`.